src/Faithlife.Analyzers/SyntaxUtility.cs

```diff
@@ -0,0 +1,108 @@
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Faithlife.Analyzers
+{
+	internal static class SyntaxUtility
+	{
+		/// <summary>
+		/// Generates an identifier name that has a high likelihood of not causing any naming conflicts.
+		/// </summary>
+		/// <param name="desiredName">The default name for the identifier.</param>
+		/// <param name="declarationLocation">A syntax location within the member in which the identifier will be declared.</param>
+		/// <param name="originalDeclaration">
+		///  An optional syntax node designating an existing variable declaration.
+		/// If specified, this node will be ignored when checking for uniqueness.
+		/// </param>
+		/// <returns>A <see cref="Microsoft.CodeAnalysis.SyntaxToken">SyntaxToken</see> for the new identifier.</returns>
+		/// <remarks>
+		/// Currently, this method takes a conservative approach to declarations within the member and pays no attention at all
+		/// to identifiers declared outside the scope of the member.
+		/// </remarks>
+		public static SyntaxToken GetHoistableIdentifier(string desiredName, SyntaxNode declarationLocation, SyntaxNode originalDeclaration = null)
+		{
+			if (desiredName is null)
+				throw new ArgumentNullException(nameof(desiredName));
+			if (declarationLocation is null)
+				throw new ArgumentNullException(nameof(declarationLocation));
+
+			var containingMember = declarationLocation.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+			if (containingMember is null)
+				throw new InvalidOperationException("Cannot declare a variable at this scope.");
+
+			// Currently, this makes no attempt to be clever about scoping rules.
+			var unavailableNames = new HashSet<string>(
+				containingMember
+					.DescendantNodes()
+					.Select(node =>
+					{
+						if (node == originalDeclaration)
+							return default;
+						if (node is ParameterSyntax parameter)
+							return parameter.Identifier;
+						if (node is TypeParameterSyntax typeParameter)
+							return typeParameter.Identifier;
+						if (node is VariableDeclaratorSyntax variableDeclarator)
+							return variableDeclarator.Identifier;
+						if (node is SingleVariableDesignationSyntax singleVariableDesignation)
+							return singleVariableDesignation.Identifier;
+						if (node is FromClauseSyntax fromClause)
+							return fromClause.Identifier;
+						if (node is LetClauseSyntax letClause)
+							return letClause.Identifier;
+						if (node is QueryContinuationSyntax queryContinuation)
+							return queryContinuation.Identifier;
+
+						return default;
+					})
+					.Select(x => x.Text)
+					.Where(x => x?.StartsWith(desiredName, StringComparison.Ordinal) ?? false),
+				StringComparer.Ordinal);
+
+			// This will also make "value" unavailable within the get accessor.
+			if (containingMember is BasePropertyDeclarationSyntax && "value".StartsWith(desiredName, StringComparison.Ordinal))
```

Comment:
Second half of the `&&` seems either unnecessary (i.e., why not just always add `"value"`) or overly broad (why is `"value"` added if `desiredName == "v"`?).